File: ios/Plugin/ThemeDetectionPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(ThemeDetectionPlugin, "ThemeDetection",
           CAP_PLUGIN_METHOD(isDarkModeActive, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnPromise);
)

File: ios/Plugin/ThemeDetectionPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ThemeDetectionPlugin)
public class ThemeDetectionPlugin: CAPPlugin {
    
    @objc func isDarkModeActive(_ call: CAPPluginCall) {
        var isDark: Bool = false
        
        if #available(iOS 13.0, *) {
            isDark = self.bridge?.userInterfaceStyle == .dark
        }
        
        call.resolve([
            "active": isDark
        ])
    }
}
